aws_lambda_powertools/tracing/tracer.py

```diff
@@ -244,7 +244,7 @@ def patch(self, modules: Tuple[str] = None):
 
     def capture_lambda_handler(
         self,
-        lambda_handler: Union[Callable[[Dict, Any], Any], Callable[[Dict, Any, Optional[Dict]], Any]] = None,
+        lambda_handler: Union[Callable[[Dict, Any], Any], Optional[Callable[[Dict, Any, Optional[Dict]], Any]]] = None,
```

Comment:
@heitorlessa haha, quiet a handful :)

Reply:
This one is massive. I can't remember why Mypy asked us to have the second Callable w/ 3 arguments - it was back when you made the first PR on Mypy. I **think** it was a customer from the Nordics who injects additional kwargs in the handler.

This deserves its own type alias OR a generic altogether: https://github.com/awslabs/aws-lambda-powertools-python/pull/517/files#diff-ac2e7c32fce52e9e17243aafe7f69a62615e44d9d3a15dabd164d16b476dc5daR22